lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -742,7 +801,15 @@ namespace Js
         Assert(this->errorObject == nullptr);
         if (this->errorObject != nullptr)
         {
-            JavascriptExceptionOperators::Throw(errorObject, scriptContext);
+            if (this->promise != nullptr)
```

Comment:
Moved assert to below in the latest iteration.

---
In reply to: [115362106](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115362106) [](ancestors = 115362106)